src/main/java/life/qbic/model/download/QbicDataDownloader.java

```diff
@@ -100,36 +103,57 @@ public void downloadRequestedFilesOfDatasets(List<String> ids, List<String> suff
         String.format(
             "%s provided openBIS identifiers have been found: %s",
             ids.size(), ids));
+    Predicate<DataSetFile> fileFilter = it -> true;
+    if (Objects.nonNull(suffixes) && !suffixes.isEmpty()) {
+      LOG.info(
+          "Applying suffix filter for suffixes: [" + String.join(" ", suffixes) + "] ...");
+      Predicate<DataSetFile> suffixFilter = file -> {
+        String fileName = getFileName(file);
+        return suffixes.stream().anyMatch(fileName::endsWith);
+      };
+      fileFilter = fileFilter.and(suffixFilter);
+    }
 
     for (String ident : ids) {
       Map<Sample, List<DataSet>> foundDataSets = qbicDataFinder.findAllDatasetsRecursive(ident);
+
       if (foundDataSets.size() > 0) {
-        LOG.info(String.format("Downloading files for identifier %s", ident));
-        try {
-          Predicate<DataSetFile> fileFilter = it -> true;
-          if (Objects.nonNull(suffixes) && !suffixes.isEmpty()) {
-            LOG.info("Applying suffix filter for suffixes: [" + String.join(" ", suffixes) + "] ...");
-            Predicate<DataSetFile> suffixFilter = file -> {
-              String fileName = getFileName(file);
-              return suffixes.stream().anyMatch(fileName::endsWith);
-            };
-            fileFilter = fileFilter.and(suffixFilter);
+        List<Sample> analyteSamples = foundDataSets.entrySet().stream()
+            .filter(it -> it.getValue().size() > 0)
+            .map(Entry::getKey)
+            .map(qbicDataFinder::searchAnalyteParent)
+            .distinct()
+            .collect(Collectors.toList());
+        Map<Sample, List<DataSet>> datasetsPerAnalyte = new HashMap<>();
+        for (Sample analyteSample : analyteSamples) {
+          datasetsPerAnalyte.put(analyteSample, new ArrayList<>());
+        }
+        for (Entry<Sample, List<DataSet>> sampleListEntry : foundDataSets.entrySet()) {
+          Sample analyte = qbicDataFinder.searchAnalyteParent(sampleListEntry.getKey());
+          if (datasetsPerAnalyte.containsKey(analyte)) {
+            datasetsPerAnalyte.get(analyte).addAll(sampleListEntry.getValue());
```

Comment:
```suggestion
        Map<Sample, List<DataSet>> datasetsPerAnalyte = new HashMap<>();
        for (Entry<Sample, List<DataSet>> sampleListEntry : foundDataSets.entrySet()) {
          List<DataSet> dataSets = sampleListEntry.getValue();
          if(dataSets.size() > 0) {
            Sample analyte = qbicDataFinder.searchAnalyteParent(sampleListEntry.getKey());
            if (!datasetsPerAnalyte.containsKey(analyte)) {
              datasetsPerAnalyte.put(analyte, new ArrayList<>());
            }
            datasetsPerAnalyte.get(analyte).addAll(dataSets);
          }
```
(only one call of searchAnalyteParent(), distinct() not necessary (any more), as empty list is only added if other datasets did not already have same analyteParent)